pkg/controller/servicebinding/servicebinder.go

```diff
@@ -298,12 +298,14 @@ func (b *serviceBinder) bind() (reconcile.Result, error) {
 
 	b.logger.Debug("Saving data on intermediary secret...")
 
-	secretObj, err := b.secret.createOrUpdate(b.envVars, b.sbr.AsOwnerReference())
+	secretObj, err := b.secret.create(b.envVars, b.sbr.AsOwnerReference())
 	if err != nil {
 		b.logger.Error(err, "On saving secret data..")
 		return b.onError(err, b.sbr, sbrStatus, nil)
 	}
+
 	sbrStatus.Secret = secretObj.GetName()
+	b.sbr.Status.Secret = sbrStatus.Secret
```

Comment:
do we really need this line? We use `sbrStatus` in this functino all over and there we have the right secret name.

Reply:
Yes.

Further in this function, we call bind method which uses `b.sbr.Status.Secret` name to set the secret name in deployment etc -> https://github.com/redhat-developer/service-binding-operator/pull/750/files#diff-e1c6315a7fe9d7e8b61a33eb162c474382845362cae5064ca666296b5456ba20R227